### PR TITLE
commitlint: build in strict mode

### PR DESCRIPTION
**Pull request by Mersho**

### CODE REVIEW

**Comment by aarani** [comment]
What's the plan here? Just adding all strict flags in TS compiler? Sorry but I hate it! This is not a spacecraft control panel. /cc @knocte

**Comment by knocte** [comment]
> What's the plan here? Just adding all strict flags in TS compiler? Sorry but I hate it! This is not a spacecraft control panel.

The plan is this: https://github.com/nblockchain/lnp2pbot/issues/2 (inspired by this: https://hackernoon.com/making-typescript-truly-strongly-typed )

**Comment by knocte** [comment]
> One way to fix this is to assign a function expression to a variable name instead:

AFAIU, your commit doesn't do this anymore.

**Comment by knocte** [comment]
@Mersho can I get an update here, what happened with this PR?

**Comment by Mersho** [comment]
> @Mersho can I get an update here, what happened with this PR?

CI of the last commit failed on "installing dependencies".
Also, the CI of the second commit failed but we don't have the logs. I'm going to investigate it now.

**Comment by Mersho** [comment]
My branch CI is green but if you want a green PR CI we need to add 'failing test' to first commit:

`please make sure that the commit message contains the 'failing test' term`

**Comment by knocte** [comment]
> My branch CI is green but if you want a green PR CI we need to add 'failing test' to first commit:

OR HOW ABOUT FIXING THE CI OF FIRST COMMIT INSTEAD OF HAVING A FAKE COMMIT MESSAGE?????

**Comment by knocte** [comment]
>> The error below occurs when a function declaration is inside a
block in strict mode when targeting ‘ES3’ or ‘ES5’. One
way to fix this is to assign a function expression to a
variable name instead:
> commitlint/plugins.ts:19:26 - error TS1251: Function declarations are not allowed inside blocks

I disagree with the way you are fixing this!!! THE POINT OF USING THE STRICT FLAG IS TO BE MORE STRICT! **If you add the strict flag but later add another flag that makes the strict flag weaker, WHAT IS THE POINT????** Just fix the goddamn compiler error!!!!!!!

**Comment by knocte** [comment]
@Mersho can you rebase this on Monday please

**Comment by knocte** [comment]
> Also print the typescript version on the CI as well.

When you have to write the word "also" in a commit message, you're already suspicious of mixing things that maybe shouldn't be mixed.

And in this case, in fact, I'm going to ask you to split that commit in two, and separate the devDependencies part to another PR please.

File: tsconfig.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "compilerOptions": {
         "lib": ["ES2021"],
         // TODO: explore replacing the below setting with just target:ES2015
-        "downlevelIteration": true
-    }
+        "downlevelIteration": true,
+        "strict": true,
```
Comment (knocte) [inline tsconfig.json:6]: @Mersho please:
1. move this to be the very first flag of the JSON file
2. I have NEVER told you to move the `tsc --version` outside of this PR. READ MY LATEST MESSAGES AGAIN TO TRY TO UNDERSTAND WHAT I SAID

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
     "compilerOptions": {
+        "strict": true,
         "lib": ["ES2021"],
         // TODO: explore replacing the below setting with just target:ES2015
-        "downlevelIteration": true
-    }
+        "downlevelIteration": true,
+    },
```
Comment (knocte) [inline tsconfig.json:7]: @Mersho these 2 lines don't need to change

**Pull request merged**